src/itype.cpp

```diff
@@ -461,3 +461,48 @@ int islot_ammo::dispersion_considering_length( units::length barrel_length ) con
     }
     return multi_lerp( lerp_points, barrel_length.value() ) + dispersion;
 }
+
+bool item_melee_damage::handle_proportional( const JsonValue &jval )
+{
+    if( jval.test_object() ) {
+        item_melee_damage rhs;
+        rhs.default_value = 1.0f;
+        rhs.deserialize( jval.get_object() );
+        for( const std::pair<const damage_type_id, float> &dt : rhs.damage_map ) {
+            const auto iter = damage_map.find( dt.first );
+            if( iter != rhs.damage_map.end() ) {
```

Comment:
Too late now but this should be plain `damage_map.end()` and not `rhs.damage_map.end()`. Fixing in #80635.